coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -257,28 +257,74 @@ class SingleClusterPlanner(dsRef: DatasetRef,
     // is optimal, if there is no overlap and even worse significant gap between the individual subqueries, retrieving
     // the entire range might be suboptimal, this still might be a better option than issuing and concatenating numerous
     // subqueries separately
-    val rangeFn = InternalRangeFunction.lpToInternalFunc(sqww.functionId)
-    var innerPeriodicSeriesPlan = sqww.innerPeriodicSeries
-    val paramsExec = materializeFunctionArgs(sqww.functionArgs, qContext)
+    var innerPlan = sqww.innerPeriodicSeries
     val window = Some(sqww.subqueryWindowMs)
-    val rangeVectorTransformer =
-      PeriodicSamplesMapper(
+    // Here the inner periodic series already has start/end/step populated
+    // in Function's toSeriesPlan(), Functions.scala subqqueryArgument() method.
+    val innerExecPlan = walkLogicalPlanTree(sqww.innerPeriodicSeries, qContext)
+    if (sqww.functionId != RangeFunctionId.AbsentOverTime) {
+      val rangeFn = InternalRangeFunction.lpToInternalFunc(sqww.functionId)
+      val paramsExec = materializeFunctionArgs(sqww.functionArgs, qContext)
+      val rangeVectorTransformer =
+        PeriodicSamplesMapper(
+          sqww.startMs, sqww.stepMs, sqww.endMs,
+          window,
+          Some(rangeFn),
+          qContext,
+          false,
+          paramsExec,
+          sqww.offsetMs,
+          false,
+          true
+        )
+      innerExecPlan.plans.foreach { p => p.addRangeVectorTransformer(rangeVectorTransformer)}
+      innerExecPlan
+    } else {
+      createAbsentOverTimePlan(innerExecPlan, innerPlan, qContext, window, sqww.offsetMs, sqww)
+    }
+  }
+
+  private def createAbsentOverTimePlan(
+    innerExecPlan: PlanResult,
+    innerPlan: PeriodicSeriesPlan,
+    qContext: QueryContext,
+    window: Option[Long],
+    offsetMs : Option[Long],
+    sqww: SubqueryWithWindowing
+  ) : PlanResult = {
+    // absent over time is essentially sum(last(series)) sent through AbsentFunctionMapper
+    innerExecPlan.plans.foreach(
+      _.addRangeVectorTransformer(PeriodicSamplesMapper(
         sqww.startMs, sqww.stepMs, sqww.endMs,
         window,
-        Some(rangeFn),
+        Some(InternalRangeFunction.lpToInternalFunc(RangeFunctionId.Last)),
         qContext,
         false,
-        paramsExec,
-        sqww.offsetMs,
-        false,
-        true
+        Seq(),
+        offsetMs,
+        false
+      ))
+    )
+    val aggregate = Aggregate(AggregationOperator.Sum, innerPlan, Nil, Seq("job"))
```

Comment:
Can you remove code duplication? I think we may have the same thing for regular absent-over-time function as well.